### PR TITLE
[Compute] `az vm create/update`: Add new parameters `--v-cpus-available`  and `--v-cpus-per-core` to support  VMSize customization

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Support for VMSizecustomization for VM https://github.com/Azure/azure-cli/issues/20655
Wait for the service to add `Microsoft.Compute/VMCustomizationPreview` to sub for testing. (done)

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -367,6 +367,8 @@ def load_arguments(self, _):
         c.argument('ephemeral_os_disk_placement', arg_type=ephemeral_placement_type,
                    help='Only applicable when used with `--size`. Allows you to choose the Ephemeral OS disk provisioning location.', is_preview=True)
         c.argument('enable_hibernation', arg_type=get_three_state_flag(), min_api='2021-03-01', help='The flag that enable or disable hibernation capability on the VM.')
+        c.argument('v_cpus_available', type=int, min_api='2021-07-01', help='Specify the number of vCPUs available for the VM')
+        c.argument('v_cpus_per_core', type=int, min_api='2021-07-01', help='Specify the vCPU to physical core ratio. Setting this property to 1 also means that hyper-threading is disabled.')
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:371]:
```suggestion
        c.argument('v_cpus_per_core', type=int, min_api='2021-07-01', help='Specify the ratio of vCPU to physical core. Setting this property to 1 also means that hyper-threading is disabled.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -406,6 +408,8 @@ def load_arguments(self, _):
                    help='Enable vTPM. It is part of trusted launch.')
         c.argument('user_data', help='UserData for the VM. It can be passed in as file or string.', completer=FilesCompleter(), type=file_type, min_api='2021-03-01')
         c.argument('enable_hibernation', arg_type=get_three_state_flag(), min_api='2021-03-01', help='The flag that enable or disable hibernation capability on the VM.')
+        c.argument('v_cpus_available', type=int, min_api='2021-07-01', help='Specify the number of vCPUs available for the VM')
+        c.argument('v_cpus_per_core', type=int, min_api='2021-07-01', help='Specify the vCPU to physical core ratio. Setting this property to 1 also means that hyper-threading is disabled.')
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:412]:
```suggestion
        c.argument('v_cpus_per_core', type=int, min_api='2021-07-01', help='Specify the ratio of vCPU to physical core. Setting this property to 1 also means that hyper-threading is disabled.')
```

**Comment by zhoxing-ms** [comment]
Since both `az vm create/update` and `az vmss create/update` support these two parameters `v_cpus_available` and `v_cpus_per_core`, we can declare these parameters here

**Comment by PARADISSEEKR** [comment]
> Since both `az vm create/update` and `az vmss create/update` support these two parameters `v_cpus_available` and `v_cpus_per_core`, we can declare these parameters here
> 
> https://github.com/Azure/azure-cli/blob/c0466a437b48e48fc456276c00fabdbd160dbeb9/src/azure-cli/azure/cli/command_modules/vm/_params.py#L1070-L1080

Their api version is not the same so I plan to implement it in another branch.

**Comment by zhoxing-ms** [comment]
> Their api version is not the same so I plan to implement it in another branch.

@PARADISSEEKR  In fact, the setting of `min_api` here is the parameter granularity. I think you need to upgrade them to the latest api-version for `v_cpus_available` and `v_cpus_per_core`.

**Review by zhoxing-ms — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 3]
```diff
@@ -498,8 +498,15 @@ def _build_storage_profile():
...
-                     'storageProfile': _build_storage_profile()}
+    vm_size_properties = {}
+    if v_cpus_available is not None:
+        vm_size_properties['vCPUsAvailable'] = v_cpus_available
+
+    if v_cpus_per_core is not None:
+        vm_size_properties['vCPUsPerCore'] = v_cpus_per_core
+
+    vm_properties = {'hardwareProfile': {'vmSize': size, 'vmSizeProperties': vm_size_properties},
+                     'networkProfile': {'networkInterfaces': nics}, 'storageProfile': _build_storage_profile()}
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:509]:
```Python
vm_properties = {'hardwareProfile': {'vmSize': size}, 'networkProfile': {'networkInterfaces': nics},
                     'storageProfile': _build_storage_profile()}

     vm_size_properties = {}
    if v_cpus_available is not None:
        vm_size_properties['vCPUsAvailable'] = v_cpus_available

    if v_cpus_per_core is not None:
        vm_size_properties['vCPUsPerCore'] = v_cpus_per_core

    if vm_size_properties
       vm_properties['hardwareProfile']['vmSizeProperties'] = vm_size_properties
```

**Pull request merged**